challenges-02.test.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => { str = str + ' The end.';
+  
+  return str;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that accepts an array and copies the first element to the end of the array. The change should be reflected in the source array that was passed in to the function. That is, the function should modify the array 'in place'.
+
+Do not use a return statement.
+
+For example:
+const a = [1, 2, 3];
+appendFirstToLast(a);
+console.log(a) prints [1, 2, 3, 1]
+------------------------------------------------------------------------------------------------ */
+
+const appendFirstToLast = (arr) => {
+  let value=arr[0];
+  arr.push(value);
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that accepts an object and an integer as arguments and adds a new property to the object called yearBorn. The value of the yearBorn property should be the integer that was passed in. 
+
+The change should be reflected in the source object that was passed in to the function. That is, the function should modify the object 'in place'.
+
+Do not use a return statement.
+
+For example:
+const octavia = { fullName: 'Octavia Estelle Butler' };
+addBirthYearProperty(octavia, 1947);
+console.log(a) prints { fullName: 'Octavia Estelle Butler', yearBorn: 1947 }
+------------------------------------------------------------------------------------------------ */
+
+const addBirthYearProperty = (obj, year) => {
+    obj.yearBorn= year;
+
+  // Solution code here...
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function that accepts an array of people objects and adds a new property called isAuthor to each object in the list. Set the value of the new property to true.
+
+The function should modify the object in place. Do not use a return statement.
+
+For example:
+const people = [{ fullName: 'Octavia Butler' }, { fullName: 'Ray Bradbury' }];
+setStatusAsAuthor(people);
+console.log(people[1].isAuthor) prints true
+------------------------------------------------------------------------------------------------ */
+
+const setStatusAsAuthor = (people) => {
+  obj.isAuthor =people;
+  allpeople=[]
```

Comment:
you have some errors in this function, what is the purpose of the empty allpeople array? Take a look at the code below and see if you can implement this
```
 people.forEach((person) => {
    person.isAuthor = true;
}
```